README.md

```diff
@@ -1,2 +1,841 @@
+
+Integrated Trust Network (ITN) is a community-built and operated network. The objective is to build a Federated Trust Layer (Core Services) and Business Automation Interoperability Infrastructure (User Agents) for ITN. The ITN Identifier is specified in accordance with the [W3C DID Core specification](https://www.w3.org/TR/did-core/).
+
+ITN Identifier = ITN DID
+
+The syntax of the identifier and accompanying data model used by the protocol is conformant with the [DID-CORE](https://www.w3.org/TR/did-core/) specification and shall be registered with the [DID-SPEC-REGISTRIES](https://www.w3.org/TR/did-spec-registries/).
+
+For a detailed read on DID and other DID method specifications, one can refer [WebOfTrustInfo - DID Primer](https://github.com/WebOfTrustInfo/rwot5-boston/blob/master/topics-and-advance-readings/did-primer.md)
+
+## Architecture Overview
+
+ITN Agent is an implementation of the [Aries Agent RFC](https://github.com/hyperledger/aries-rfcs/blob/main/concepts/0004-agents/README.md)
+with a flexible plugin architecture.
+
+Depending on which plugins and [DIDComm protocols](https://didcomm.org/search/?page=1) you use, your instance of ITN Agent can
+perform a variety of roles:
+
+* Create and manage cryptographic keys
+* Create and manage Decentralized Identifiers
+* Issue and verify Verifiable Credentials (VCs) and Presentations (VPs)
+* Communicate with other Agents over [DIDComm](https://identity.foundation/didcomm-messaging/spec/#discover-features-protocol-20)
+
+Agent provides a common interface for plugins to expand its functionality. It is designed to be modular and extensible,
+so you can add new protocols, transports, and managers to fit your needs.
+
+The agent was written in [TypeScript](https://www.typescriptlang.org/) and runs natively in Node.js, Browsers, and React Native.
+
+## ITN SDK
+
+![ITN_SDK](https://github.com/itn-trust/itn-did-spec/assets/18353464/6135b073-c085-41fd-8ea4-f80e64e0485d)
+
+One of the protocols is Identity Protocol provides methods to manage DIDs.
+One of the managers is DID Document Manager provide a set of functions to manage DID Documents.
+
+`Conirm with Umed & Andreas - here I am only providing necessary information related to DIDs & DID Document. Is that okay?`
+
+## Conformance
+
+The keywords MAY, MUST, MUST NOT, RECOMMENDED, SHOULD, and SHOULD NOT in this document are to be interpreted... TBD `Not sure if required`
+
+## Terminology
+
+1. ITN - Integrated Trust Network
+2. EDV - Encrypted Data Vault 
+
+## DID ITN Method Specification
+
+### Format
+
+The ITN DID is derived as the sha3-256 hash of a random salt of sufficient entropy.
+
+The format of ITN DID conform to the [W3C DID Core specification](https://www.w3.org/TR/did-core/). It consists of `did:itn` prefix followed by `sha3-256(random salt)`. The W3C DID Core specification does not specify how a DID is generated, and leaves it up to the implementation provided it ensures uniqueness to a high degree of confidence.
```

Comment:
the DID `identifier` creation is more complicated then simple hashing:
1. Generate random 32 bytes seed (for the Agent `primary` DID the seed is derived from the master key)
2. Get `sha512` hash of  the generated seed
3. Generate `Ed25519` keypair using hashed seed as a derivation seed - the generated keypair will be a Recovery keypair
4. Create DID identifier by encoding first 16 bytes of the Recovery keypair's public key into `Base58` format
5. Prefix the generated identifier with `did:itn`

Here is a pseudocode for the generation process:

```ts
const seed = randomBytes(32)
const seedHash = sha512(seed)
const keyPair = Ed25519.keyPair(seedHash)
const identifier = Base58.encode(keyPair.publicKey.slice(0, 16))
const did = `did:itn:${identifier}`
```

Recovery keypair stored in DID controller's wallet. The public key also stored in CAS. All subsequent operations on the DID (update, revoke, recover) are signed by **private** recovery key and verified by **public** recovery key. Verification process also includes the DID identifier validation - having the public recovery key, the verifier repeats steps 4 and 5 on his side and makes sure the DID matches the recovery key.

Reply:
@bparth24 this is just for explanation purposes, it doesn't mean we should include such implementation details.

Reply:
Noted & Updated the Format section content.